### PR TITLE
a small wrapper for seeking in C

### PR DESCRIPTION
**Pull request by clutchski**
a small wrapper which (hopefully) eliminates a bit of overhead when seeking for a particular key.

### CODE REVIEW

File: seek_to.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
+
+#include "leveldb/c.h"
+#include <stdbool.h>
+
+typedef struct {
+    unsigned char valid;
+    unsigned char equal;
+} SeekResult;
+
+SeekResult leveldb_iter_seek_to(leveldb_iterator_t* iter, const char* k, size_t klen) {
```
Comment (jmoiron) [inline seek_to.h]: code in a .h file is kinda poor form due to the way the preprocessor works;  check out https://github.com/DataDog/czlib/blob/master/fastzlib.h and https://github.com/DataDog/czlib/blob/master/fastzlib.c I think it's pretty straightforward to give the implementation a .c file and have things JustWork.

File: seek_to.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package levigo
+
+// #include "seek_to.h"
+import "C"
+import "unsafe"
+
+type SeekResult struct {
+	Valid bool
+	Equal bool
```
Comment (jmoiron) [inline seek_to.go]: It'd be nice to get the value in here eventually.  Cut the cgo overhead even more.

File: seek_to.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    // do the seek
+    leveldb_iter_seek(iter, k, klen);
+
+    SeekResult sr;
+    sr.valid = leveldb_iter_valid(iter);
+
+    if (sr.valid != 0) {
+        size_t olen;
+        const char* ok = leveldb_iter_key(iter, &olen);
```
Comment (jmoiron) [inline seek_to.h]: assuming this is idiom in leveldb code but `ok` is a confusing var name for this give how it's used in go

File: seek_to.h (hunk 1 above)
Comment (clutchski) [inline seek_to.h]: yea just sloth will fix

File: seek_to.go (hunk 2 above)
Comment (clutchski) [inline seek_to.go]: done

**Pull request merged**